### PR TITLE
fix(ingest) Athena: db filter was not applied

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/sql/athena.py [hunk 1]
```diff
@@ -33,10 +33,22 @@ def get_sql_alchemy_url(self):
 
 
 class AthenaSource(SQLAlchemySource):
+    config: AthenaConfig
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/sql/athena.py:36]: Delete this. The `config` is already an instance member of SQLAlchemySource, and will get correctly initialized via lines 44->39.

File: metadata-ingestion/src/datahub/ingestion/source/sql/athena.py [hunk 2]
```diff
@@ -33,10 +33,22 @@ def get_sql_alchemy_url(self):
...
     def __init__(self, config, ctx):
         super().__init__(config, ctx, "athena")
 
     @classmethod
     def create(cls, config_dict, ctx):
         config = AthenaConfig.parse_obj(config_dict)
         return cls(config, ctx)
+
+    # It seems like database/schema filter in the connection string does not work and this to work around that
+    def get_schema_names(self, inspector):
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/sql/athena.py:47]: missing return type annotation.

File: metadata-ingestion/src/datahub/ingestion/source/sql/athena.py [hunk 3]
```diff
@@ -33,10 +33,22 @@ def get_sql_alchemy_url(self):
...
+        schemas = inspector.get_schema_names()
+        if self.config.database:
+            for schema in schemas:
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/sql/athena.py:50]: 50-53
```python
return [schema in schemas if schema == self.config.database]
```

**Review by rslanka — CHANGES_REQUESTED** [review]
Needs changes.

**Review by shirshanka — APPROVED** [review]
LGTM

**Review by shirshanka — APPROVED** [review]

**Pull request merged**